src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -17,6 +17,7 @@ public sealed class ResourceReader : IDisposable
         private Stream _stream;    // backing store we're reading from.
         private long _nameSectionOffset;  // Offset to name section of file.
         private long _dataSectionOffset;  // Offset to Data section of file.
+        private long _resourceStreamStart;  // beginning of the resource stream, which could be part of a longer file stream
```

Comment:
Can you add a test where this will end up being something other than 0?  Otherwise it's difficult to know whether all of the additions / subtractions being done are correct.